agent/consul/fsm/fsm.go

```diff
@@ -218,3 +202,36 @@ func (c *FSM) Restore(old io.ReadCloser) error {
 	stateOld.Abandon()
 	return nil
 }
+
+// ReadSnapshot decodes each message type and utilizes the handler function to
+// process each message type individually
+func ReadSnapshot(r io.Reader, handler func(header *SnapshotHeader, msg structs.MessageType, dec *codec.Decoder) error) error {
+	// Create a decoder
+	dec := codec.NewDecoder(r, structs.MsgpackHandle)
+
+	// Read in the header
+	var header SnapshotHeader
+	if err := dec.Decode(&header); err != nil {
+		return err
+	}
+
+	// Populate the new state
+	msgType := make([]byte, 1)
+	for {
+		// Read the message type
+		_, err := r.Read(msgType)
+		if err == io.EOF {
+			break
```

Comment:
```suggestion
			return nil
```
(trivial) Since this isn't inside of some other function, break and return are equivalent here